src/nemos/solvers.py

```diff
@@ -0,0 +1,725 @@
+            leaves having the same sized first dimension (corresponding to the number of data points).
+            For GLMs these are:
+                X : DESIGN_INPUT_TYPE
+                    Input data.
+                y : jnp.ndarray
+                    Output data.
+
+
+        Returns
+        -------
+        OptStep
+            xs :
+                Parameters after taking one step defined in the inner loop of Prox-SVRG.
+            state :
+                Updated state.
+
+        Raises
+        ------
+        ValueError
+            The parameter update needs a value for the full gradient at the anchor point, which needs the full data
+            to be calculated and is expected to be stored in state.df_xs. So if state.df_xs is None, a ValueError is raised.
+        """
+        if state.df_xs is None:
+            raise ValueError(
+                "Full gradient at the anchor point (state.df_xs) has to be set. "
+                + "Try passing init_full_gradient=True to ProxSVRG.init_state or GLM.initialize_solver."
+            )
+        return self._update_on_batch(current_params, state, prox_lambda, *args)
+
+    @partial(jit, static_argnums=(0,))
+    def _update_on_batch(
+        self,
+        current_params: Pytree,
+        state: SVRGState,
+        prox_lambda: Union[float, None],
+        *args,
+    ) -> OptStep:
+        """
+        Update parameters given a mini-batch of data and increment iteration/epoch number in state.
+
+        Note that this method doesn't update state.xs, state.df_xs, that has to be done outside.
+
+        Parameters
+        ----------
+        current_params :
+            Parameters at the end of the previous update, used as the starting point for the current update.
+        state :
+            Optimizer state at the end of the previous update.
+            Needs to have the current anchor point (xs) and the gradient at the anchor point (df_xs) already set.
+        prox_lambda :
+            Regularization strength.
+        args:
+            Positional arguments passed to loss function `fun` and its gradient (e.g. `fun(current_params, *args)`),
+            most likely input and output data.
+            They are expected to be Pytrees with arrays or FeaturePytree as their leaves, with all of their
+            leaves having the same sized first dimension (corresponding to the number of data points).
+            For GLMs these are:
+                X : DESIGN_INPUT_TYPE
+                    Input data.
+                y : jnp.ndarray
+                    Output data.
+
+        Returns
+        -------
+        OptStep
+            xs :
+                Parameters after taking one step defined in the inner loop of Prox-SVRG.
+            state :
+                Updated state.
+        """
+        # NOTE this doesn't update state.xs, state.df_xs, that has to be done outside
```

Comment:
this is already in the docstring, so can be removed

Reply:
Removed it.